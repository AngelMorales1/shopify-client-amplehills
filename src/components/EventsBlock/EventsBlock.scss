@import 'settings', 'typography', 'utils';

.EventsBlock {
  @extend .text-madison-blue;

  &--white {
    @extend .bg-white;
  }

  &--yellow {
    @extend .bg-deep-yellow;
  }

  &--pink {
    @extend .bg-peach-pink;
  }

  &__button-container {
    flex-direction: column;

    @media ($medium-up) {
      flex-direction: row;
    }
  }

  &__button-container {
    flex-direction: column;

    @media ($medium-up) {
      flex-direction: row;
    }
  }
}
