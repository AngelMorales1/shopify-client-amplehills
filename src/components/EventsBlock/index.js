import React, { Component } from 'react';
import PropTypes from 'prop-types';
import marked from 'marked';
import cx from 'classnames';
import get from 'utils/get';
import EventCard from 'components/EventCard';
import EventTypes from 'constants/EventTypes';

import { Button } from 'components/base';
import styles from './EventsBlock.scss';

class EventsBlock extends Component {
  state = {
    selectedEvents: [],
    buttonLabels: [],
    activeFilter: ''
  };

  componentDidMount() {
    const events = get(this, 'props.block.fields.events', {});
    const selectedEvents = Object.values(events).length
      ? this.getCustomEvents()
      : this.getAllEvents();

    const buttonLabels = this.getLocationButtonLabel(selectedEvents);

    this.setState({
      selectedEvents,
      buttonLabels,
      activeFilter: buttonLabels[0]
    });
  }

  filterIsActive = filter => {
    return this.state.activeFilter === filter;
  };

  cardIsActive = event => {
    const locationFilterButtonIsOn = get(
      this,
      'props.block.fields.locationFilterButton',
      false
    );

    if (locationFilterButtonIsOn) {
      const eventLocation = get(event, 'locationTitle', '');

      return this.state.activeFilter.split(':')[0] === eventLocation;
    }

    return true;
  };

  getLocationButtonLabel = selectedEvents => {
    const locationButtonLabel = selectedEvents.reduce((acc, cur) => {
      const location = get(cur, 'locationTitle', '');

      if (location) {
        acc[location] ? acc[location]++ : (acc[location] = 1);
      }

      return acc;
    }, {});

    return Object.keys(locationButtonLabel).map(
      key => `${key}: ${locationButtonLabel[key]}`
    );
  };

  getAllEvents = () => {
    const events = get(this, 'props.events', {});
    const allEvents = Object.values(events);
    const blockEventType = get(this, 'props.block.fields.eventType', '');

    return allEvents.filter(event => {
      const eventType = get(event, 'eventType', '');

      if (!blockEventType || blockEventType === EventTypes.ALL_EVENTS) {
        return event;
      }

      return eventType === blockEventType;
    });
  };

  getCustomEvents = () => {
    const events = get(this, 'props.events', {});
    const customEvents = get(this, 'props.block.fields.events', []);
    const allEvents = Object.values(events);

    return customEvents.map(customEvent => {
      const id = get(customEvent, 'sys.id', '');
      return allEvents.find(event => event.contentfulId === id);
    });
  };

  render() {
    const { z, block, setRef } = this.props;
    const { buttonLabels, selectedEvents } = this.state;
    const fields = get(block, 'fields', {});
    const isDripOn = get(fields, 'drip', false);
    const colorClass = `EventsBlock--${get(fields, 'backgroundColor', '')}`;
    const title = get(fields, 'title', '');
    const text = get(fields, 'text', '');
    const locationFilterButtonIsOn = get(fields, 'locationFilterButton', false);

    return (
      <div
        ref={refBlock => setRef(refBlock)}
        style={{ zIndex: z }}
        className={cx(
          styles[colorClass],
          'flex flex-column items-center justify-center',
          {
            drip: isDripOn
          }
        )}
      >
        <div className="px2 text-container-width center">
          <h2 className="block-headline mt4 mb3">{title}</h2>
          <div
            dangerouslySetInnerHTML={{
              __html: marked(text)
            }}
            className="markdown-block"
          />
        </div>
        {locationFilterButtonIsOn ? (
          <div
            className={cx(
              styles['EventsBlock__button-container'],
              'flex flex-wrap items-center justify-center mt3'
            )}
          >
            {buttonLabels.map((label, i) => {
              const color = this.filterIsActive(label)
                ? 'clear-madison-blue-border'
                : 'madison-blue';

              return (
                <Button
                  className="m1"
                  color={color}
                  variant="primary-small"
                  key={label}
                  label={label}
                  onClick={() =>
                    this.setState({
                      activeFilter: label
                    })
                  }
                />
              );
            })}
          </div>
        ) : null}
        <div className="w100 flex flex-column items-center my3 px2">
          {selectedEvents.map((event, i) => {
            return event ? (
              <EventCard
                active={this.cardIsActive(event)}
                key={get(event, 'contentfulId', i)}
                event={event}
              />
            ) : null;
          })}
        </div>
      </div>
    );
  }
}

EventsBlock.propTypes = {
  z: PropTypes.number,
  block: PropTypes.shape({
    fields: PropTypes.shape({
      backgroudColor: '',
      title: PropTypes.string,
      drip: PropTypes.bool,
      eventType: PropTypes.string,
      text: PropTypes.string,
      events: PropTypes.array,
      locationFilterButton: PropTypes.bool
    })
  }),
  events: PropTypes.array,
  setRef: PropTypes.func
};

EventsBlock.defaultProps = {
  z: 0,
  block: {
    fields: {
      backgroudColor: 'white',
      title: '',
      drip: false,
      eventType: '',
      text: '',
      events: [],
      locationFilterButton: false
    }
  },
  events: [],
  setRef: () => {}
};

export default EventsBlock;
