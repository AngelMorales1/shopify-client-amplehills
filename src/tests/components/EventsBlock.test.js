import React from 'react';
import { shallow } from 'enzyme';

import EventsBlock from 'components/EventsBlock';

it('renders without data', () => {
  const component = shallow(<EventsBlock />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <EventsBlock
      z={0}
      block={{
        fields: {
          backgroudColor: 'foo',
          title: 'foo',
          drip: false,
          eventType: 'foo',
          text: 'foo',
          events: [],
          locationFilterButton: false
        }
      }}
      events={[
        {
          id: 'foo',
          blockCardText: 'foo',
          eventType: 'foo',
          image: 'foo',
          locationTitle: 'foo',
          title: 'foo',
          datesAndTimes: [
            {
              uuid: 'foo',
              Date: 'foo',
              Time: 'foo'
            }
          ]
        }
      ]}
      setRef={() => console.log('foo')}
    />
  );

  expect(component).toMatchSnapshot();
});
