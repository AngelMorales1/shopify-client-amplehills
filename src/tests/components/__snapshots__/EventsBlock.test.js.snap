// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventsBlock
    block={
        Object {
            "fields": Object {
              "backgroudColor": "foo",
              "drip": false,
              "eventType": "foo",
              "events": Array [],
              "locationFilterButton": false,
              "text": "foo",
              "title": "foo",
            },
          }
    }
    events={
        Array [
            Object {
              "blockCardText": "foo",
              "datesAndTimes": Array [
                Object {
                  "Date": "foo",
                  "Time": "foo",
                  "uuid": "foo",
                },
              ],
              "eventType": "foo",
              "id": "foo",
              "image": "foo",
              "locationTitle": "foo",
              "title": "foo",
            },
          ]
    }
    setRef={[Function]}
    z={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="px2 text-container-width center"
>
          <h2
                    className="block-headline mt4 mb3"
          >
                    foo
          </h2>
          <div
                    className="markdown-block"
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": "<p>foo</p>
                              ",
                                      }
                    }
          />
</div>,
        null,
        <div
          className="w100 flex flex-column items-center my3 px2"
>
          <EventCard
                    active={true}
                    event={
                              Object {
                                        "blockCardText": "foo",
                                        "datesAndTimes": Array [
                                          Object {
                                            "Date": "foo",
                                            "Time": "foo",
                                            "uuid": "foo",
                                          },
                                        ],
                                        "eventType": "foo",
                                        "id": "foo",
                                        "image": "foo",
                                        "locationTitle": "foo",
                                        "title": "foo",
                                      }
                    }
          />
</div>,
      ],
      "className": "flex flex-column items-center justify-center",
      "style": Object {
        "zIndex": 0,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="block-headline mt4 mb3"
>
              foo
</h2>,
            <div
              className="markdown-block"
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "<p>foo</p>
                            ",
                                        }
              }
/>,
          ],
          "className": "px2 text-container-width center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "foo",
              "className": "block-headline mt4 mb3",
            },
            "ref": null,
            "rendered": "foo",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "markdown-block",
              "dangerouslySetInnerHTML": Object {
                "__html": "<p>foo</p>
",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <EventCard
              active={true}
              event={
                            Object {
                                          "blockCardText": "foo",
                                          "datesAndTimes": Array [
                                            Object {
                                              "Date": "foo",
                                              "Time": "foo",
                                              "uuid": "foo",
                                            },
                                          ],
                                          "eventType": "foo",
                                          "id": "foo",
                                          "image": "foo",
                                          "locationTitle": "foo",
                                          "title": "foo",
                                        }
              }
/>,
          ],
          "className": "w100 flex flex-column items-center my3 px2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
<<<<<<< HEAD
            "key": "0",
=======
            "key": "foo",
>>>>>>> Add EventBlock test
            "nodeType": "function",
            "props": Object {
              "active": true,
              "event": Object {
                "blockCardText": "foo",
                "datesAndTimes": Array [
                  Object {
                    "Date": "foo",
                    "Time": "foo",
                    "uuid": "foo",
                  },
                ],
                "eventType": "foo",
                "id": "foo",
                "image": "foo",
                "locationTitle": "foo",
                "title": "foo",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="px2 text-container-width center"
>
            <h2
                        className="block-headline mt4 mb3"
            >
                        foo
            </h2>
            <div
                        className="markdown-block"
                        dangerouslySetInnerHTML={
                                    Object {
                                                "__html": "<p>foo</p>
                                    ",
                                              }
                        }
            />
</div>,
          null,
          <div
            className="w100 flex flex-column items-center my3 px2"
>
            <EventCard
                        active={true}
                        event={
                                    Object {
                                                "blockCardText": "foo",
                                                "datesAndTimes": Array [
                                                  Object {
                                                    "Date": "foo",
                                                    "Time": "foo",
                                                    "uuid": "foo",
                                                  },
                                                ],
                                                "eventType": "foo",
                                                "id": "foo",
                                                "image": "foo",
                                                "locationTitle": "foo",
                                                "title": "foo",
                                              }
                        }
            />
</div>,
        ],
        "className": "flex flex-column items-center justify-center",
        "style": Object {
          "zIndex": 0,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="block-headline mt4 mb3"
>
                foo
</h2>,
              <div
                className="markdown-block"
                dangerouslySetInnerHTML={
                                Object {
                                                "__html": "<p>foo</p>
                                ",
                                              }
                }
/>,
            ],
            "className": "px2 text-container-width center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "foo",
                "className": "block-headline mt4 mb3",
              },
              "ref": null,
              "rendered": "foo",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "markdown-block",
                "dangerouslySetInnerHTML": Object {
                  "__html": "<p>foo</p>
",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <EventCard
                active={true}
                event={
                                Object {
                                                "blockCardText": "foo",
                                                "datesAndTimes": Array [
                                                  Object {
                                                    "Date": "foo",
                                                    "Time": "foo",
                                                    "uuid": "foo",
                                                  },
                                                ],
                                                "eventType": "foo",
                                                "id": "foo",
                                                "image": "foo",
                                                "locationTitle": "foo",
                                                "title": "foo",
                                              }
                }
/>,
            ],
            "className": "w100 flex flex-column items-center my3 px2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
<<<<<<< HEAD
              "key": "0",
=======
              "key": "foo",
>>>>>>> Add EventBlock test
              "nodeType": "function",
              "props": Object {
                "active": true,
                "event": Object {
                  "blockCardText": "foo",
                  "datesAndTimes": Array [
                    Object {
                      "Date": "foo",
                      "Time": "foo",
                      "uuid": "foo",
                    },
                  ],
                  "eventType": "foo",
                  "id": "foo",
                  "image": "foo",
                  "locationTitle": "foo",
                  "title": "foo",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventsBlock
    block={
        Object {
            "fields": Object {
              "backgroudColor": "white",
              "drip": false,
              "eventType": "",
              "events": Array [],
              "locationFilterButton": false,
              "text": "",
              "title": "",
            },
          }
    }
    events={
        Array [
            Object {
              "blockCardText": "",
              "datesAndTimes": Array [
                Object {
                  "Date": "",
                  "Time": "",
                  "uuid": "",
                },
              ],
              "eventType": "",
              "id": "",
              "image": "",
              "locationTitle": "",
              "title": "",
            },
          ]
    }
    setRef={[Function]}
    z={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="px2 text-container-width center"
>
          <h2
                    className="block-headline mt4 mb3"
          />
          <div
                    className="markdown-block"
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": "",
                                      }
                    }
          />
</div>,
        null,
        <div
          className="w100 flex flex-column items-center my3 px2"
>
          <EventCard
                    active={true}
                    event={
                              Object {
                                        "blockCardText": "",
                                        "datesAndTimes": Array [
                                          Object {
                                            "Date": "",
                                            "Time": "",
                                            "uuid": "",
                                          },
                                        ],
                                        "eventType": "",
                                        "id": "",
                                        "image": "",
                                        "locationTitle": "",
                                        "title": "",
                                      }
                    }
          />
</div>,
      ],
      "className": "flex flex-column items-center justify-center",
      "style": Object {
        "zIndex": 0,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="block-headline mt4 mb3"
/>,
            <div
              className="markdown-block"
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "",
                                        }
              }
/>,
          ],
          "className": "px2 text-container-width center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "block-headline mt4 mb3",
            },
            "ref": null,
            "rendered": "",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "markdown-block",
              "dangerouslySetInnerHTML": Object {
                "__html": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <EventCard
              active={true}
              event={
                            Object {
                                          "blockCardText": "",
                                          "datesAndTimes": Array [
                                            Object {
                                              "Date": "",
                                              "Time": "",
                                              "uuid": "",
                                            },
                                          ],
                                          "eventType": "",
                                          "id": "",
                                          "image": "",
                                          "locationTitle": "",
                                          "title": "",
                                        }
              }
/>,
          ],
          "className": "w100 flex flex-column items-center my3 px2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
<<<<<<< HEAD
            "key": "0",
=======
            "key": undefined,
>>>>>>> Add EventBlock test
            "nodeType": "function",
            "props": Object {
              "active": true,
              "event": Object {
                "blockCardText": "",
                "datesAndTimes": Array [
                  Object {
                    "Date": "",
                    "Time": "",
                    "uuid": "",
                  },
                ],
                "eventType": "",
                "id": "",
                "image": "",
                "locationTitle": "",
                "title": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="px2 text-container-width center"
>
            <h2
                        className="block-headline mt4 mb3"
            />
            <div
                        className="markdown-block"
                        dangerouslySetInnerHTML={
                                    Object {
                                                "__html": "",
                                              }
                        }
            />
</div>,
          null,
          <div
            className="w100 flex flex-column items-center my3 px2"
>
            <EventCard
                        active={true}
                        event={
                                    Object {
                                                "blockCardText": "",
                                                "datesAndTimes": Array [
                                                  Object {
                                                    "Date": "",
                                                    "Time": "",
                                                    "uuid": "",
                                                  },
                                                ],
                                                "eventType": "",
                                                "id": "",
                                                "image": "",
                                                "locationTitle": "",
                                                "title": "",
                                              }
                        }
            />
</div>,
        ],
        "className": "flex flex-column items-center justify-center",
        "style": Object {
          "zIndex": 0,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="block-headline mt4 mb3"
/>,
              <div
                className="markdown-block"
                dangerouslySetInnerHTML={
                                Object {
                                                "__html": "",
                                              }
                }
/>,
            ],
            "className": "px2 text-container-width center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "block-headline mt4 mb3",
              },
              "ref": null,
              "rendered": "",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "markdown-block",
                "dangerouslySetInnerHTML": Object {
                  "__html": "",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <EventCard
                active={true}
                event={
                                Object {
                                                "blockCardText": "",
                                                "datesAndTimes": Array [
                                                  Object {
                                                    "Date": "",
                                                    "Time": "",
                                                    "uuid": "",
                                                  },
                                                ],
                                                "eventType": "",
                                                "id": "",
                                                "image": "",
                                                "locationTitle": "",
                                                "title": "",
                                              }
                }
/>,
            ],
            "className": "w100 flex flex-column items-center my3 px2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
<<<<<<< HEAD
              "key": "0",
=======
              "key": undefined,
>>>>>>> Add EventBlock test
              "nodeType": "function",
              "props": Object {
                "active": true,
                "event": Object {
                  "blockCardText": "",
                  "datesAndTimes": Array [
                    Object {
                      "Date": "",
                      "Time": "",
                      "uuid": "",
                    },
                  ],
                  "eventType": "",
                  "id": "",
                  "image": "",
                  "locationTitle": "",
                  "title": "",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
